llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -1771,6 +1776,13 @@ m_SelectCst(const Cond &C) {
   return m_Select(C, m_ConstantInt<L>(), m_ConstantInt<R>());
 }
 
+/// Match Select(C, LHS, RHS) or Select(C, RHS, LHS)
+template <typename Cond, typename LHS, typename RHS>
+inline ThreeOps_match<Cond, LHS, RHS, Instruction::Select, true>
+m_c_Select(const Cond &C, const LHS &L, const RHS &R) {
+  return ThreeOps_match<Cond, LHS, RHS, Instruction::Select, true>(C, L, R);
```

Comment:
I think this needs to hide the condition, or add something to indicate the condition needs to be inverted. You can't safely directly use the condition if it's swapped